src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,246 @@
+    using astropy's sigma clipping functions
+    """
+
+    sigma_clipping_max_sigma = Int(
+        default_value=4,
+        help="Maximal value for the sigma clipping outlier removal",
+    ).tag(config=True)
+    sigma_clipping_iterations = Int(
+        default_value=5,
+        help="Number of iterations for the sigma clipping outlier removal",
+    ).tag(config=True)
+
+    def __call__(
+        self, dl1_table, masked_pixels_of_sample=None, col_name="image"
+    ) -> list:
+
+        # in python 3.12 itertools.batched can be used
+        image_chunks = (
+            dl1_table[col_name].data[i : i + self.sample_size]
+            for i in range(0, len(dl1_table[col_name].data), self.sample_size)
+        )
+        time_chunks = (
+            dl1_table["time"][i : i + self.sample_size]
+            for i in range(0, len(dl1_table["time"]), self.sample_size)
+        )
+
+        # Calculate the statistics from a sequence of images
+        stats_list = []
+        for images, times in zip(image_chunks, time_chunks):
+            stats_list.append(
+                self._sigmaclipping_extraction(images, times, masked_pixels_of_sample)
+            )
+        return stats_list
+
+    def _sigmaclipping_extraction(
+        self, images, times, masked_pixels_of_sample
+    ) -> StatisticsContainer:
+
+        # ensure numpy array
+        masked_images = np.ma.array(images, mask=masked_pixels_of_sample)
+
+        # median of the event images
+        image_median = np.ma.median(masked_images, axis=-1)
+
+        # mean, median, and std over the sample per pixel
+        max_sigma = self.sigma_clipping_max_sigma
+        pixel_mean, pixel_median, pixel_std = sigma_clipped_stats(
+            masked_images,
+            sigma=max_sigma,
+            maxiters=self.sigma_clipping_iterations,
+            cenfunc="mean",
+            axis=0,
+        )
+
+        # mask pixels without defined statistical values
+        pixel_mean = np.ma.array(pixel_mean, mask=np.isnan(pixel_mean))
+        pixel_median = np.ma.array(pixel_median, mask=np.isnan(pixel_median))
+        pixel_std = np.ma.array(pixel_std, mask=np.isnan(pixel_std))
+
+        unused_values = np.abs(masked_images - pixel_mean) > (max_sigma * pixel_std)
+
+        # only warn for values discard in the sigma clipping, not those from before
+        outliers = unused_values & (~masked_images.mask)
+
+        # add outliers identified by sigma clipping for following operations
+        masked_images.mask |= unused_values
+
+        # median of the median over the camera
+        median_of_pixel_median = np.ma.median(pixel_median, axis=1)
+
+        # median of the std over the camera
+        median_of_pixel_std = np.ma.median(pixel_std, axis=1)
+
+        # std of the std over camera
+        std_of_pixel_std = np.ma.std(pixel_std, axis=1)
+
+        # outliers from median
+        image_deviation = pixel_median - median_of_pixel_median[:, np.newaxis]
```

Comment:
this is absolute deviation, in the docstring you say relative. Please clarify and update accordingly

Reply:
same @FrancaCassol

Reply:
sorry, of which docstring are we speaking? I can not find it. In any case, the point is that the deviation is evaluated with respect to the median value , not as absolute value